consensus/src/network.rs

```diff
@@ -39,11 +40,30 @@ use futures::{
     stream::{select, select_all},
     SinkExt, Stream, StreamExt,
 };
+use serde::{de::DeserializeOwned, Serialize};
 use std::{
     mem::{discriminant, Discriminant},
     time::Duration,
 };
 
+pub trait ConsensusMessageTrait: Sized + Clone + Serialize + DeserializeOwned {
```

Comment:
i personally prefer shorter names, there's conventions in other languages to name it like TConsensusMessage, TDagMessage